### PR TITLE
Error summary at the end of restore

### PR DESCRIPTION
**Pull request by BrennanConroy**
#1376

### CODE REVIEW

**Comment by msotclas** [comment]
Hi __@BrennanConroy__, I'm your friendly neighborhood Microsoft Open Technologies, Inc. Pull Request Bot (You can call me MSOTBOT). Thanks for your contribution!
    <p>
        It looks like you're a Microsoft contributor (Brennan Conroy). If you're full-time, we DON'T require a Contribution License Agreement. If you are a vendor, please DO sign the electronic Contribution License Agreement. It will take 2 minutes and there's no faxing! https://cla2.msopentech.com.
    </p>

TTYL, MSOTBOT;

File: src/Microsoft.Framework.PackageManager/Restore/RestoreCommand.cs [hunk 1]
```diff
@@ -30,6 +30,7 @@ public RestoreCommand(IApplicationEnvironment env)
             FileSystem = new PhysicalFileSystem(Directory.GetCurrentDirectory());
             MachineWideSettings = new CommandLineMachineWideSettings();
             ScriptExecutor = new ScriptExecutor();
+            FailedPackages = new Dictionary<string, List<string>>();
```
Comment (dougbu) [inline src/Microsoft.Framework.PackageManager/Restore/RestoreCommand.cs]: add explicit comparer (`StringComparer.Ordinal`) here and on line 58 ?  (might not be as common in this repo)

File: src/Microsoft.Framework.PackageManager/Restore/RestoreCommand.cs [hunk 2]
```diff
@@ -142,6 +144,15 @@ public RestoreCommand(IApplicationEnvironment env)
                     Reports.Information.WriteLine(string.Format("Total time {0}ms", sw.ElapsedMilliseconds));
                 }
 
+                foreach(var json in FailedPackages)
```
Comment (dougbu) [inline src/Microsoft.Framework.PackageManager/Restore/RestoreCommand.cs]: - `json` -> `error`, `package`, or `failedPackage`
- nit: space after `foreach`

File: src/Microsoft.Framework.PackageManager/Restore/RestoreCommand.cs [hunk 3]
```diff
@@ -142,6 +144,15 @@ public RestoreCommand(IApplicationEnvironment env)
...
+                {
+                    Reports.Error.WriteLine("Errors in {0}".Red().Bold(), json.Key);
+                    foreach(var error in json.Value)
```
Comment (dougbu) [inline src/Microsoft.Framework.PackageManager/Restore/RestoreCommand.cs]: nit: guess I don't really like `error`. suggest `message` or `errorMessage` here.

File: src/Microsoft.Framework.PackageManager/Restore/RestoreCommand.cs [hunk 4]
```diff
@@ -243,7 +256,9 @@ public RestoreCommand(IApplicationEnvironment env)
                          node.LibraryRange.VersionRange != null &&
                          missingItems.Add(node.LibraryRange))
                     {
-                        Reports.Error.WriteLine(string.Format("Unable to locate {0} >= {1}", node.LibraryRange.Name.Red().Bold(), node.LibraryRange.VersionRange));
+                        var errorMessage = string.Format("Unable to locate {0} >= {1}", node.LibraryRange.Name.Red().Bold(), node.LibraryRange.VersionRange);
```
Comment (dougbu) [inline src/Microsoft.Framework.PackageManager/Restore/RestoreCommand.cs]: nit: a few long lines get longer in this PR

**Comment by dougbu** [comment]
why are errors that occur in script (e.g. `prerestore`) execution not included in the summary?

**Comment by dougbu** [comment]
:watch:

**Comment by BrennanConroy** [comment]
Updated
The RestoreForInstall function is going away in #1370

**Comment by dougbu** [comment]
:shipit:

**Pull request closed without merging**